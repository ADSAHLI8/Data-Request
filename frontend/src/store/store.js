import { configureStore } from '@reduxjs/toolkit'
import DataRequestReducer from '../features/DataRequest/dataRequestSlice'
import InsertionConditionReducer from '../features/InsertionCondition/insertionConditionSlice'
import {requestApi} from '../features/requestApi'


 const store = configureStore({
    reducer: {
        dataRequest : DataRequestReducer,
        insertionCondition : InsertionConditionReducer,
        [requestApi.reducerPath] : requestApi.reducer
    },
    middleware : (getDefaultMiddleware) => getDefaultMiddleware().concat(requestApi.middleware)
    
})
export default store