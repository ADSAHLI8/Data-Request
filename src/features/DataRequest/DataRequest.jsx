import React from 'react'
import { SelectButton } from 'primereact/selectbutton';
import entityOptions from "../../data/entityOptions";
import countryOptions from "../../data/countriesOptions";
import ispOptions from '../../data/ispOptions';
import CreatableSelect from 'react-select/creatable';
import { InputText } from "primereact/inputtext";
import { InputSwitch } from "primereact/inputswitch";
import { Card } from 'primereact/card';
import countries2Options from '../../data/countries2Options';
import "../../css/dataRequest.css"
import { useDispatch, useSelector } from 'react-redux';
import {changeEntity,changeCountry,changeCountry2,changeISP,changeDomain,changeDashboardName,changeIs_Amazon,
  changeIs_Paypal,changeIs_Fedex,changeIs_Insurance,changeIs_Hardbounce} from './dataRequestSlice'

/**
 * @component
 * 
 * @returns{JSX.Element} 
 */


const DataRequest = () => {
  /** Defining a state for each form element */
   const Entity = useSelector((state)=>state.dataRequest.Entity)
   const Country = useSelector((state)=>state.dataRequest.Country)
   const Country2 = useSelector((state)=>state.dataRequest.Country2)
   const ISP = useSelector((state)=>state.dataRequest.ISP)
   const Domain = useSelector((state)=>state.dataRequest.Domain)
   const DashboardName = useSelector((state)=>state.dataRequest.DashboardName)
   const Is_Amazon = useSelector((state)=>state.dataRequest.Is_Amazon)
   const Is_Paypal = useSelector((state)=>state.dataRequest.Is_Paypal)
   const Is_Fedex = useSelector((state)=>state.dataRequest.Is_Fedex)
   const Is_Insurance = useSelector((state)=>state.dataRequest.Is_Insurance)
   const Is_Hardbounce = useSelector((state)=>state.dataRequest.Is_Hardbounce)
   /** Calling use dispatch */
   const dispatch = useDispatch()
   /** Return data request Form */
  return (
    <div className='r'>
      <Card title="Entity">
            <SelectButton value={Entity} onChange={(e)=>dispatch(changeEntity(e.target.value))} className='entity'  optionLabel="name" name='entity' options={entityOptions} multiple />
      </Card>
      <Card title="Country">
    <div className="card flex justify-content-center">
            <SelectButton value={Country} onChange={(e)=>dispatch(changeCountry(e.target.value))} optionLabel="name" className='countries' name='country'  options={countryOptions} multiple />
    </div>
    <div className="select-group">
        <CreatableSelect
          isMulti
          name="country2"
          className="select"
          options={countries2Options}
          classNamePrefix="select"
          placeholder = "Select Another Country"
          onChange={(e)=>dispatch(changeCountry2(e))}
          value={Country2}
          />
    </div>
      </Card>
      <div className='cards'>

      <Card>
    <div className="label-inline" id='isp-label'>
        ISP
    </div>
        <CreatableSelect
          isMulti
          id='isp-select'
          name="isp"
          className="select"
          options={ispOptions}
          classNamePrefix="select"
          placeholder = "Select ISP"
          onChange={(e)=>dispatch(changeISP(e))}
          value={ISP}
          />
    <div className="label-inline " id='domain-label' >
        Domain
    </div>
        <InputText 
          id='domain-input'
          name="domain"
          className="input"
          placeholder = "Input Domain"
          onChange={(e)=>dispatch(changeDomain(e.target.value))}
          value={Domain}
          />
    <div className="label" id='dashboard-name-label'>
        Dashboard Name
    </div>
      <InputText 
      id='dashboard-name-input'
      className='input'
      value={DashboardName} 
      name="dashboardName"
      placeholder = "Input Dashboard Name"
      onChange={(e)=>dispatch(changeDashboardName(e.target.value))} />
      </Card>
    <Card >
      <div className='switch'>
        <div className="switch-label" id='isAmazon'>
         Is_Amazon
        </div>
        <InputSwitch checked={Is_Amazon} name ="isAmazon" className='switch-input'  onChange={(e)=>dispatch(changeIs_Amazon(e.target.value))} />
      </div>
      <div className='switch'>
       <div className="switch-label" id='isPaypal'>
        Is_Paypal
       </div>
        <InputSwitch checked={Is_Paypal} name ="isPaypal" className='switch-input' onChange={(e)=>dispatch(changeIs_Paypal(e.target.value))} />
      </div>
      <div className='switch'>
       <div className="switch-label" id='isFedex'>
        Is_Fedex
      </div>
        <InputSwitch checked={Is_Fedex} name ="isFedex" className='switch-input' onChange={(e)=>dispatch(changeIs_Fedex(e.target.value))} />
      </div>
      <div className="switch">
       <div className="switch-label" id='isInsurance'>
        Is_Insurance
      </div>
        <InputSwitch checked={Is_Insurance} name ="isInsurance" className='switch-input' onChange={(e)=>dispatch(changeIs_Insurance(e.target.value))} />
      </div>
      <div className="switch">
       <div className="switch-label" id='isHardbounce'>
        Is_Hardbounce
      </div>
      <InputSwitch checked={Is_Hardbounce} name ="isHardbounce" className='switch-input' onChange={(e)=>dispatch(changeIs_Hardbounce(e.target.value))} />
      </div>
        </Card>      
      </div>
        
    </div>
  )
}
/** Export Component */
export default DataRequest