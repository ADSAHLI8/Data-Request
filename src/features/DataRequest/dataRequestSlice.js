import { createSlice } from "@reduxjs/toolkit"
  /**   Create initial state */
const initialState = {
    Entity : '',
    Country : '',
    Country2 : '',
    ISP : '',
    Domain : '',
    DashboardName:'',
    Is_Amazon : false,
    Is_Paypal: false,
    Is_Fedex:false,
    Is_Insurance:false,
    Is_Hardbounce:false
}
/** Create data request slice and actions */
const dataRequestSlice = createSlice({
    name : 'dataRequest',
    initialState,
    reducers:{
        changeEntity: (state,action) => {
            state.Entity=action.payload
        },
        changeCountry: (state,action) => {
            state.Country=action.payload
        },
        changeCountry2: (state,action) => {
            state.Country2=action.payload
        },
        changeISP: (state,action) => {
            state.ISP=action.payload
        },
        changeDomain: (state,action) => {
            state.Domain=action.payload
        },
        changeDashboardName: (state,action) => {
            state.DashboardName=action.payload
        },
        changeIs_Amazon: (state,action) => {
            state.Is_Amazon=action.payload
        },
        changeIs_Paypal: (state,action) => {
            state.Is_Paypal=action.payload
        },
        changeIs_Fedex: (state,action) => {
            state.Is_Fedex=action.payload
        },
        changeIs_Insurance: (state,action) => {
            state.Is_Insurance=action.payload
        },
        changeIs_Hardbounce: (state,action) => {
            state.Is_Hardbounce=action.payload
        },
        reset: (state) =>{
            state.Entity = ''
            state.Country = ''
            state.Country2 = ''
            state.ISP = ''
            state.Domain = ''
            state.DashboardName=''
            state.Is_Amazon = false
            state.Is_Paypal= false
            state.Is_Fedex=false
            state.Is_Insurance=false
            state.Is_Hardbounce=false
        },
    },
})
/** Export reducer and actions */
export default dataRequestSlice.reducer
export const {changeEntity,changeCountry,changeCountry2,changeISP,changeDomain,changeDashboardName,changeIs_Amazon,
    changeIs_Paypal,changeIs_Fedex,changeIs_Insurance,changeIs_Hardbounce,reset}= dataRequestSlice.actions