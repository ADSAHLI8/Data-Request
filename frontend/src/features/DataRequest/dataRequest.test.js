import { Provider } from "react-redux";
import DataRequest from "./DataRequest";
import { render,screen } from "@testing-library/react";
import '@testing-library/jest-dom'; // For matchers like 'toBeInTheDocument'
import store from "../../store/store";
import { PrimeReactProvider } from "primereact/api";

 const MockedDataRequest = () => {
    return(
        <Provider store={store}>
            <PrimeReactProvider>
              <DataRequest />
            </PrimeReactProvider>
        </Provider>
    )
}

describe('Data Request',() =>{
    it('should render buttons',async ()=>{
        render (<MockedDataRequest />);
        const buttonElements= screen.getAllByRole('button')
        expect(buttonElements.length).toBe(24)
    })
    it('should render Labels',async ()=>{
        render (<MockedDataRequest />);
        const selectElement= screen.getByText('Entity')
        expect(selectElement).toBeInTheDocument()
        const selectElement1= screen.getByText('Country')
        expect(selectElement1).toBeInTheDocument()
        const selectElement2= screen.getByText('ISP')
        expect(selectElement2).toBeInTheDocument()
        const selectElement3= screen.getByText('Domain')
        expect(selectElement3).toBeInTheDocument()
        const selectElement4= screen.getByText('Dashboard Name')
        expect(selectElement4).toBeInTheDocument()
        const selectElement5= screen.getByText('Is_Amazon')
        expect(selectElement5).toBeInTheDocument()
        const selectElement6= screen.getByText('Is_Paypal')
        expect(selectElement6).toBeInTheDocument()
        const selectElement7= screen.getByText('Is_Insurance')
        expect(selectElement7).toBeInTheDocument()
        const selectElement8= screen.getByText('Is_Fedex')
        expect(selectElement8).toBeInTheDocument()
        const selectElement9= screen.getByText('Is_Hardbounce')
        expect(selectElement9).toBeInTheDocument()
    })
    it('should render Country Select',async ()=>{
        render (<MockedDataRequest />);
        const selectElement= screen.getByText('Select Another Country')
        expect(selectElement).toBeInTheDocument()
    })
    it('should render isp Select',async ()=>{
        render (<MockedDataRequest />);
        const selectElement= screen.getByText('Select ISP')
        expect(selectElement).toBeInTheDocument()
    })
    it('should render domain input',async ()=>{
        render (<MockedDataRequest />);
        const inputElement= screen.getByPlaceholderText('Input Domain')
        expect(inputElement).toBeInTheDocument()
    })
    it('should render dashboard name input',async ()=>{
        render (<MockedDataRequest />);
        const inputElement= screen.getByPlaceholderText('Input Dashboard Name')
        expect(inputElement).toBeInTheDocument()
    })
    it('should render switches',async ()=>{
        render (<MockedDataRequest />);
        const switchElements= screen.getAllByRole('switch')
        expect(switchElements.length).toBe(5)
    })

})