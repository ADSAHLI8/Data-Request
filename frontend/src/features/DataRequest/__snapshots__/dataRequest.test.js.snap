// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Request should render correctly 1`] = `
<div
  className="r"
>
  <div
    className="p-card p-component"
    data-pc-name="card"
    data-pc-section="root"
    id={null}
    style={Object {}}
  >
    <div
      className="p-card-body"
      data-pc-section="body"
    >
      <div
        className="p-card-title"
        data-pc-section="title"
      >
        Entity
      </div>
      <div
        className="p-card-content"
        data-pc-section="content"
      >
        <div
          className="entity p-selectbutton p-button-group p-component"
          data-pc-name="selectbutton"
          data-pc-section="root"
          id={null}
          name="entity"
          role="group"
          style={Object {}}
        >
          <div
            aria-disabled={false}
            aria-label="itm"
            aria-pressed={false}
            className="p-button p-component"
            data-pc-section="button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <span
              className="p-button-label p-c"
              data-pc-section="label"
            >
              itm
            </span>
          </div>
          <div
            aria-disabled={false}
            aria-label="advertix"
            aria-pressed={false}
            className="p-button p-component"
            data-pc-section="button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="-1"
          >
            <span
              className="p-button-label p-c"
              data-pc-section="label"
            >
              advertix
            </span>
          </div>
          <div
            aria-disabled={false}
            aria-label="adsglory1"
            aria-pressed={false}
            className="p-button p-component"
            data-pc-section="button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="-1"
          >
            <span
              className="p-button-label p-c"
              data-pc-section="label"
            >
              adsglory1
            </span>
          </div>
          <div
            aria-disabled={false}
            aria-label="adsglory3"
            aria-pressed={false}
            className="p-button p-component"
            data-pc-section="button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="-1"
          >
            <span
              className="p-button-label p-c"
              data-pc-section="label"
            >
              adsglory3
            </span>
          </div>
          <div
            aria-disabled={false}
            aria-label="wikileads"
            aria-pressed={false}
            className="p-button p-component"
            data-pc-section="button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="-1"
          >
            <span
              className="p-button-label p-c"
              data-pc-section="label"
            >
              wikileads
            </span>
          </div>
          <div
            aria-disabled={false}
            aria-label="avalon"
            aria-pressed={false}
            className="p-button p-component"
            data-pc-section="button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="-1"
          >
            <span
              className="p-button-label p-c"
              data-pc-section="label"
            >
              avalon
            </span>
          </div>
          <div
            aria-disabled={false}
            aria-label="ilead"
            aria-pressed={false}
            className="p-button p-component"
            data-pc-section="button"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="-1"
          >
            <span
              className="p-button-label p-c"
              data-pc-section="label"
            >
              ilead
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="p-card p-component"
    data-pc-name="card"
    data-pc-section="root"
    id={null}
    style={Object {}}
  >
    <div
      className="p-card-body"
      data-pc-section="body"
    >
      <div
        className="p-card-title"
        data-pc-section="title"
      >
        Country
      </div>
      <div
        className="p-card-content"
        data-pc-section="content"
      >
        <div
          className="card flex justify-content-center"
        >
          <div
            className="countries p-selectbutton p-button-group p-component"
            data-pc-name="selectbutton"
            data-pc-section="root"
            id={null}
            name="country"
            role="group"
            style={Object {}}
          >
            <div
              aria-disabled={false}
              aria-label="EUROPE"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                EUROPE
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="US"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                US
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="UK"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                UK
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="ES"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                ES
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="NO"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                NO
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="FR"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                FR
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="NL"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                NL
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="DE"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                DE
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="AU"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                AU
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="CA"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                CA
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="BE"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                BE
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="PL"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                PL
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="SE"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                SE
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="BR"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                BR
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="NZ"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                NZ
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="DK"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                DK
              </span>
            </div>
            <div
              aria-disabled={false}
              aria-label="FI"
              aria-pressed={false}
              className="p-button p-component"
              data-pc-section="button"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="-1"
            >
              <span
                className="p-button-label p-c"
                data-pc-section="label"
              >
                FI
              </span>
            </div>
          </div>
        </div>
        <div
          className="select-group"
        >
          <div
            className="select css-b62m3t-container"
            onKeyDown={[Function]}
          >
            <span
              className="css-1f43avz-a11yText-A11yText"
              id="react-select-16-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              className="css-1f43avz-a11yText-A11yText"
              role="log"
            />
            <div
              className="select__control css-13cymwt-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className="select__value-container select__value-container--is-multi css-hlgwow"
              >
                <div
                  className="select__placeholder css-1jqq78o-placeholder"
                  id="react-select-16-placeholder"
                >
                  Select Another Country
                </div>
                <div
                  className="select__input-container css-19bb58m"
                  data-value=""
                >
                  <input
                    aria-activedescendant=""
                    aria-autocomplete="list"
                    aria-describedby="react-select-16-placeholder"
                    aria-expanded={false}
                    aria-haspopup={true}
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    className="select__input"
                    disabled={false}
                    id="react-select-16-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    role="combobox"
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "color": "inherit",
                        "font": "inherit",
                        "gridArea": "1 / 2",
                        "label": "input",
                        "margin": 0,
                        "minWidth": "2px",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "100%",
                      }
                    }
                    tabIndex={0}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="select__indicators css-1wy0on6"
              >
                <span
                  className="select__indicator-separator css-1u9des2-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className="select__indicator select__dropdown-indicator css-1xc3v61-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-tj5bde-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div>
              <input
                name="country2"
                type="hidden"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="cards"
  >
    <div
      className="p-card p-component"
      data-pc-name="card"
      data-pc-section="root"
      id={null}
      style={Object {}}
    >
      <div
        className="p-card-body"
        data-pc-section="body"
      >
        <div
          className="p-card-content"
          data-pc-section="content"
        >
          <div
            className="label-inline"
            id="isp-label"
          >
            ISP
          </div>
          <div
            className="select css-b62m3t-container"
            id="isp-select"
            onKeyDown={[Function]}
          >
            <span
              className="css-1f43avz-a11yText-A11yText"
              id="react-select-17-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              className="css-1f43avz-a11yText-A11yText"
              role="log"
            />
            <div
              className="select__control css-13cymwt-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className="select__value-container select__value-container--is-multi css-hlgwow"
              >
                <div
                  className="select__placeholder css-1jqq78o-placeholder"
                  id="react-select-17-placeholder"
                >
                  Select ISP
                </div>
                <div
                  className="select__input-container css-19bb58m"
                  data-value=""
                >
                  <input
                    aria-activedescendant=""
                    aria-autocomplete="list"
                    aria-describedby="react-select-17-placeholder"
                    aria-expanded={false}
                    aria-haspopup={true}
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    className="select__input"
                    disabled={false}
                    id="react-select-17-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    role="combobox"
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "color": "inherit",
                        "font": "inherit",
                        "gridArea": "1 / 2",
                        "label": "input",
                        "margin": 0,
                        "minWidth": "2px",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "100%",
                      }
                    }
                    tabIndex={0}
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="select__indicators css-1wy0on6"
              >
                <span
                  className="select__indicator-separator css-1u9des2-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  className="select__indicator select__dropdown-indicator css-1xc3v61-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-tj5bde-Svg"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div>
              <input
                name="isp"
                type="hidden"
                value=""
              />
            </div>
          </div>
          <div
            className="label-inline "
            id="domain-label"
          >
            Domain
          </div>
          <input
            className="input p-inputtext p-component"
            data-pc-name="inputtext"
            data-pc-section="root"
            id="domain-input"
            name="domain"
            onBeforeInput={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            onKeyDown={[Function]}
            onPaste={[Function]}
            placeholder="Input Domain"
            value=""
          />
          <div
            className="label"
            id="dashboard-name-label"
          >
            Dashboard Name
          </div>
          <input
            className="input p-inputtext p-component"
            data-pc-name="inputtext"
            data-pc-section="root"
            id="dashboard-name-input"
            name="dashboardName"
            onBeforeInput={[Function]}
            onChange={[Function]}
            onInput={[Function]}
            onKeyDown={[Function]}
            onPaste={[Function]}
            placeholder="Input Dashboard Name"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="p-card p-component"
      data-pc-name="card"
      data-pc-section="root"
      id={null}
      style={Object {}}
    >
      <div
        className="p-card-body"
        data-pc-section="body"
      >
        <div
          className="p-card-content"
          data-pc-section="content"
        >
          <div
            className="switch"
          >
            <div
              className="switch-label"
              id="isAmazon"
            >
              Is_Amazon
            </div>
            <div
              aria-checked={false}
              className="switch-input p-inputswitch p-component"
              data-p-disabled={false}
              data-p-highlight={false}
              data-pc-name="inputswitch"
              data-pc-section="root"
              id={null}
              role="checkbox"
              style={Object {}}
            >
              <input
                aria-checked={false}
                checked={false}
                className="p-inputswitch-input"
                data-pc-section="input"
                disabled={false}
                id={null}
                name="isAmazon"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="switch"
                tabIndex={null}
                type="checkbox"
              />
              <span
                className="p-inputswitch-slider"
                data-pc-section="slider"
              />
            </div>
          </div>
          <div
            className="switch"
          >
            <div
              className="switch-label"
              id="isPaypal"
            >
              Is_Paypal
            </div>
            <div
              aria-checked={false}
              className="switch-input p-inputswitch p-component"
              data-p-disabled={false}
              data-p-highlight={false}
              data-pc-name="inputswitch"
              data-pc-section="root"
              id={null}
              role="checkbox"
              style={Object {}}
            >
              <input
                aria-checked={false}
                checked={false}
                className="p-inputswitch-input"
                data-pc-section="input"
                disabled={false}
                id={null}
                name="isPaypal"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="switch"
                tabIndex={null}
                type="checkbox"
              />
              <span
                className="p-inputswitch-slider"
                data-pc-section="slider"
              />
            </div>
          </div>
          <div
            className="switch"
          >
            <div
              className="switch-label"
              id="isFedex"
            >
              Is_Fedex
            </div>
            <div
              aria-checked={false}
              className="switch-input p-inputswitch p-component"
              data-p-disabled={false}
              data-p-highlight={false}
              data-pc-name="inputswitch"
              data-pc-section="root"
              id={null}
              role="checkbox"
              style={Object {}}
            >
              <input
                aria-checked={false}
                checked={false}
                className="p-inputswitch-input"
                data-pc-section="input"
                disabled={false}
                id={null}
                name="isFedex"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="switch"
                tabIndex={null}
                type="checkbox"
              />
              <span
                className="p-inputswitch-slider"
                data-pc-section="slider"
              />
            </div>
          </div>
          <div
            className="switch"
          >
            <div
              className="switch-label"
              id="isInsurance"
            >
              Is_Insurance
            </div>
            <div
              aria-checked={false}
              className="switch-input p-inputswitch p-component"
              data-p-disabled={false}
              data-p-highlight={false}
              data-pc-name="inputswitch"
              data-pc-section="root"
              id={null}
              role="checkbox"
              style={Object {}}
            >
              <input
                aria-checked={false}
                checked={false}
                className="p-inputswitch-input"
                data-pc-section="input"
                disabled={false}
                id={null}
                name="isInsurance"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="switch"
                tabIndex={null}
                type="checkbox"
              />
              <span
                className="p-inputswitch-slider"
                data-pc-section="slider"
              />
            </div>
          </div>
          <div
            className="switch"
          >
            <div
              className="switch-label"
              id="isHardbounce"
            >
              Is_Hardbounce
            </div>
            <div
              aria-checked={false}
              className="switch-input p-inputswitch p-component"
              data-p-disabled={false}
              data-p-highlight={false}
              data-pc-name="inputswitch"
              data-pc-section="root"
              id={null}
              role="checkbox"
              style={Object {}}
            >
              <input
                aria-checked={false}
                checked={false}
                className="p-inputswitch-input"
                data-pc-section="input"
                disabled={false}
                id={null}
                name="isHardbounce"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="switch"
                tabIndex={null}
                type="checkbox"
              />
              <span
                className="p-inputswitch-slider"
                data-pc-section="slider"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
